src/main/java/org/jabref/logic/bibtexkeypattern/BibtexKeyGenerator.java

```diff
@@ -24,11 +24,12 @@
 public class BibtexKeyGenerator extends BracketedPattern {
     /*
      * All single characters that we can use for extending a key to make it unique.
+     * KEY_CHARACTERS_CAUSING_PARSING_ERRORS is a subset of KEY_ILLEGAL_CHARACTERS
      */
     public static final String APPENDIX_CHARACTERS = "abcdefghijklmnopqrstuvwxyz";
     private static final Logger LOGGER = LoggerFactory.getLogger(BibtexKeyGenerator.class);
-    private static final String KEY_ILLEGAL_CHARACTERS = "{}(),\\\"-#~^:'`ʹ";
-    private static final String KEY_UNWANTED_CHARACTERS = "{}(),\\\"-";
+    private static final String KEY_ILLEGAL_CHARACTERS = "{}(),=\\\"#%~'`ʹ";
```

Comment:
Could you please add a comment here referencing https://tex.stackexchange.com/a/408548/9075?

Proposal:

```java
// Source of illegal characters: https://tex.stackexchange.com/a/408548/9075
```